commerce_coordinator/apps/commercetools/clients.py

```diff
@@ -369,6 +369,28 @@ def get_payment_by_transaction_interaction_id(self, interaction_id: str) -> CTPa
         logger.info(f"[CommercetoolsAPIClient] - Attempting to find payment with interaction ID {interaction_id}")
         return self.base_client.payments.query(where=f'transactions(interactionId="{interaction_id}")').results[0]
 
+    def get_product_variant_by_program(self, product_id: str) -> Optional[CTProductVariant]:
+        """
+        Fetches a program from Commercetools.
+        Args:
+            product_id: The ID of the program (bundle) to fetch.
+        Returns:
+            CTProductVariant if found, None otherwise.
+        """
+        start_time = datetime.datetime.now()
+        try:
+            product_projection = self.base_client.product_projections.get_by_key(product_id)
```

Comment:
This endpoint returns a JSON response. We should check if the retrieved product is returned in the JSON response. See the example for single course variant search result: https://github.com/edx/commerce-coordinator/blob/2670c6303f319c79725747b8e08c7baccb70a206/commerce_coordinator/apps/commercetools/clients.py#L382